service/history/historyEngine.go

```diff
@@ -1392,6 +1423,9 @@ Update_History_Loop:
 		// the history and try the operation again.
 		var updateErr error
 		if continueAsNewBuilder != nil {
+			if msBuilder.GetContinueAsNew() != nil {
```

Comment:
i think this will always be non nil after checking the continueAsNewBuilder != nil
ref: https://github.com/uber/cadence/pull/915/files#diff-01045b0895719962d1f13440e8795d1bR2416